src/org/jitsi/impl/neomedia/rtp/TransportCCEngine.java

```diff
@@ -599,4 +560,136 @@ private MediaStream getMediaStream()
+            extends SinglePacketTransformerAdapter
+            implements TransformEngine
+    {
+        /**
+         * Ctor.
+         */
+        private EgressEngine()
+        {
+            super(RTPPacketPredicate.INSTANCE);
+        }
+
+        /**
+         * {@inheritDoc}
+         * <p></p>
+         * If the transport-cc extension is configured, update the
+         * transport-wide sequence number (adding a new extension if one doesn't
+         * exist already).
+         */
+        @Override
+        public RawPacket transform(RawPacket pkt)
+        {
+            if (extensionId != -1)
+            {
+                RawPacket.HeaderExtension ext
+                    = pkt.getHeaderExtension((byte) extensionId);
+                if (ext == null)
+                {
+                    ext = pkt.addExtension((byte) extensionId, 2);
+                }
+
+                int seq = outgoingSeq.getAndIncrement() & 0xffff;
+                RTPUtils.writeShort(
+                    ext.getBuffer(),
+                    ext.getOffset() + 1,
+                    (short) seq);
+
+                if (logger.isDebugEnabled())
+                {
+                    logger.debug("rtp_seq=" + pkt.getSequenceNumber()
+                            + ",pt=" + pkt.getPayloadType()
+                            + ",tcc_seq=" + seq);
+                }
+
+                synchronized (sentPacketsSyncRoot)
+                {
+                    sentPacketDetails.put(seq, new PacketDetail(
+                                pkt.getLength(),
+                                System.currentTimeMillis()));
+                }
+            }
+            return pkt;
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public PacketTransformer getRTPTransformer()
+        {
+            return this;
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public PacketTransformer getRTCPTransformer()
+        {
+            return null;
+        }
+    }
+
+
+    /**
+     * Handles incoming RTP packets for this {@link TransportCCEngine}.
+     */
+    public class IngressEngine
+            extends SinglePacketTransformerAdapter
+            implements TransformEngine
+    {
+        /**
+         * Ctor.
+         */
+        private IngressEngine()
+        {
+            super(RTPPacketPredicate.INSTANCE);
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public RawPacket reverseTransform(RawPacket pkt)
+        {
+            if (extensionId != -1)
+            {
+                RawPacket.HeaderExtension he
+                    = pkt.getHeaderExtension((byte) extensionId);
+                if (he != null && he.getExtLength() == 2)
+                {
+                    int seq = RTPUtils.readUint16AsInt(
+                            he.getBuffer(), he.getOffset() + 1);
+                    packetReceived(
```

Comment:
I know the pt is only used in logging, but it's implicitly converted from a byte to an int here